ui/desktop/src/hooks/useToggleToolOutput.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useCallback } from 'react';
+
+/**
+ * Custom hook to handle Ctrl+R hotkey for toggling full tool output display
+ * This hook adds a global keyboard listener that expands all truncated tool outputs
+ */
+export function useToggleToolOutput() {
+  const toggleAllToolOutputs = useCallback(() => {
+    // Find all collapsed tool argument values and expand them
+    const collapsedButtons = document.querySelectorAll(
+      'button[aria-label="Expand value"], button:has(.expand-icon), button[data-tool-arg-key]'
+    );
+
+    // Also find any elements with truncate classes that contain tool arguments
+    const truncatedElements = document.querySelectorAll(
+      '.truncate button[onClick*="toggleKey"], .truncate.min-w-0 button[onClick*="toggleKey"]'
+    );
+
+    // Dispatch click events to expand all collapsed items
+    [...collapsedButtons, ...truncatedElements].forEach((element) => {
+      if (element instanceof HTMLElement) {
+        element.click();
+      }
+    });
```

Comment:
this should stick to a React approach of setting state, not invoking .click on dom elements